### PR TITLE
Set operational status for guarded FRUs

### PR DESCRIPTION
**Pull request by SunnySrivastava1984**
The script checks for guarded FRUs and sets the operational status as false for those FRUs.
The script does not operate in chassis on condition.

### CODE REVIEW

**Comment by SunnySrivastava1984** [comment]
jenkins -> run tests please

File: scripts/oem/set-guarded-fru-leds.sh [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+if [ "${current_chassis_status}" = \
+"\"xyz.openbmc_project.State.Chassis.PowerState.On\"" ]; then
+    echo "Current chassis power state is , $current_chassis_status . \
+    Exit set-guarded-fru-leds.sh \
+    script successfully without setting operational status for guarded FRUs."
+    exit 0
+fi
+
+#Extract list of DIMMs and Cores that has been guarded. This is required to set
```
Comment (jinuthomas) [inline scripts/oem/set-guarded-fru-leds.sh:18]: Extract list of DIMMs and Cores that has been guarded
Extract list of FRUs that has been guarded

File: scripts/oem/set-guarded-fru-leds.sh [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+#operation status property for those DIMMs  and Cores.
```
Comment (jinuthomas) [inline scripts/oem/set-guarded-fru-leds.sh:19]: DIMMs and Cores.
FRUs.

File: scripts/oem/set-guarded-fru-leds.sh [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        #check for guarded dimms
+        echo "$line2" | grep "dimm" >/dev/null
+        rc=$?
+        if [ $rc -eq 0 ]; then
+            busctl set-property xyz.openbmc_project.Inventory.Manager \
+                "$line2" xyz.openbmc_project.State.Decorator.OperationalStatus \
+                Functional b false;
+        fi
+
+        #check for guarded processors not cores.
```
Comment (jinuthomas) [inline scripts/oem/set-guarded-fru-leds.sh:42]: not cores. ?

File: meson.build [hunk 4]
```diff
@@ -166,6 +166,12 @@ install_data(
     install_dir: get_option('bindir')
 )
 
+install_data(
```
Comment (jinuthomas) [inline meson.build:169]: is two install_data required ?

**Review by jinuthomas — COMMENTED** [review]
minor nits, coding fine

File: scripts/oem/set-guarded-fru-leds.sh (hunk 1 above)
Comment (SunnySrivastava1984) [reply scripts/oem/set-guarded-fru-leds.sh:18]: Are there any other FRUs that can be guarded?

File: scripts/oem/set-guarded-fru-leds.sh (hunk 2 above)
Comment (SunnySrivastava1984) [reply scripts/oem/set-guarded-fru-leds.sh:19]: We are explicitly changing operational status of only DIMMs and Processors in this script not FRUs. Will update Cores to processors?

**Review by spinler — COMMENTED** [review]
No additional comments from me.

File: scripts/oem/set-guarded-fru-leds.sh [hunk 5]
```diff
@@ -0,0 +1,59 @@
+#!/bin/sh
+
+# This shell script sets operational status to false for guarded FRUs.
+# It is being done only for DIMM and PROCs, as gaurd is done for dimm, proc and cores
```
[the remark this answers is not in the document]
Comment (SunnySrivastava1984) [reply scripts/oem/set-guarded-fru-leds.sh:4]: What are skippable FRUs?

[the remark this answers is not in the document]
Comment (SunnySrivastava1984) [reply scripts/oem/set-guarded-fru-leds.sh:4]: > may I know the reason why you are explicitly checking for dimm and procs?
Mentioned in the comments at top.

[the remark this answers is not in the document]
Comment (SunnySrivastava1984) [reply scripts/oem/set-guarded-fru-leds.sh:4]: > can you also add why this script is required

I will add that in the commit message.

**Review by jinuthomas — APPROVED** [review]

File: scripts/oem/set-guarded-fru-leds.sh (hunk 5 above)
[the remark this answers is not in the document]
Comment (PriyangaRamasamy) [reply scripts/oem/set-guarded-fru-leds.sh]: >What are skippable FRUs?
I meant the skippable entries like cores

**Review by PriyangaRamasamy — APPROVED** [review]

**Review by jinuthomas — APPROVED** [review]

**Pull request merged**